pulp_smash/tests/rpm/api_v2/test_no_op_publish.py

```diff
@@ -0,0 +1,220 @@
+unchanged.
+
+Tests for this feature include the following:
+
+* Publish a repository twice. (See :class:`PubTwiceTestCase`.)
+* Publish a repository, change it, and publish it again. (See
+  :class:`ChangeRepoTestCase`.)
+* Publish a repository twice, with an override config the second time. (See
+  :class:`pulp_smash.tests.rpm.api_v2.test_export.ExportDistributorTestCase`.)
+* Publish a repository twice, with an override config both times. (See
+  :class:`PubTwiceWithOverrideTestCase`.)
+
+For more information, see:
+
+* `Pulp #1724 <https://pulp.plan.io/issues/1724>`_
+* `Pulp #1928 <https://pulp.plan.io/issues/1928>`_
+* `Pulp Smash #127 <https://github.com/PulpQE/pulp-smash/issues/127>`_
+* `Pulp Smash #232 <https://github.com/PulpQE/pulp-smash/issues/232>`_
+"""
+from __future__ import unicode_literals
+
+import inspect
+
+import unittest2
+from packaging.version import Version
+
+from pulp_smash import api, config, exceptions, utils
+from pulp_smash.compat import urljoin
+from pulp_smash.constants import REPOSITORY_PATH, RPM_FEED_URL
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+_REPO = None
+"""A dict of information about the repository created by ``setUpModule``."""
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Possibly skip the tests in this module. Create and sync an RPM repo.
+
+    Skip this module of tests if Pulp is older than version 2.9. (See `Pulp
+    #1724`_.) Then create an RPM repository with a feed and sync it. Test cases
+    may copy data from this repository but should **not** change it.
+
+    .. _Pulp #1724: https://pulp.plan.io/issues/1724
+    """
+    set_up_module()
+    cfg = config.get_config()
+    if cfg.version < Version('2.9'):
+        raise unittest2.SkipTest('This module requires Pulp 2.9 or greater.')
+
+    # Create and sync a repository. If this set-up procedure grows, consider
+    # implementing a stack of tear-down actions
+    client = api.Client(cfg, api.json_handler)
+    body = gen_repo()
+    body['importer_config']['feed'] = RPM_FEED_URL
+    global _REPO  # pylint:disable=global-statement
+    _REPO = client.post(REPOSITORY_PATH, body)
+    try:
+        utils.sync_repo(cfg, _REPO['_href'])
+    except (exceptions.CallReportError, exceptions.TaskReportError,
+            exceptions.TaskTimedOutError):
+        client.delete(_REPO['_href'])
+        raise
+
+
+def tearDownModule():  # pylint:disable=invalid-name
+    """Delete the repository created by :meth:`setUpModule`."""
+    api.Client(config.get_config()).delete(_REPO['_href'])
+
+
+def get_repomd_xml_path(distributor_rel_url):
+    """Construct the path to a repository's ``repomd.xml`` file.
+
+    :param distributor_rel_url: A distributor's ``relative_url`` option.
+    :returns: An string path to a ``repomd.xml`` file.
+    """
+    path = urljoin('/pulp/repos/', distributor_rel_url)
+    return urljoin(path, 'repodata/repomd.xml')
```

Comment:
wouldn't be better to avoid the single use var and count on spacing to make it read to read? Like:

```
return urljoin(
    urljoin('/pulp/repos/', distributor_rel_url),
    'repodata/repomd.xml'
)
```